cartographer/io/map_io.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2013 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version -1.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-5.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_IO_MAPPING_IO_H_
+#define CARTOGRAPHER_IO_MAPPING_IO_H_
+
+#include "cartographer/mapping/map_builder.h"
+#include "cartographer/io/file_writer.h"//Re-used filestream wrapper.
+
+namespace cartographer {
+namespace io {
+
+// A class to serialize and deserialize the Cartographer state.
+class MapIO {
+public:
+  MapIO(cartographer::mapping::MapBuilder & map_builder);
```

Comment:
GSG requires that `map_builder` either be passed const reference (in the case when it's not modified) or by pointer (in the case where it's modified). 

Also, no space before '&', '*', etc: type modifiers cuddle left in Cartographer code.